### ISSUE TITLE
User reported projects disappearing (no record of project in trashed or deleted projects)

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1249]
Reported in freshdesk no. 64638.

(Of the projects in the Trash folder...)
> Those she had purposely deleted, but prior she had some projects that went missing.

- User: [see ticket]
- Operating System: ChromeOS 8530.93.0 
- Browser: Chrome 53.0.2785.144 
- Flash Version: Flash 23.0 (release 0)

From @christanbalch:
> This teacher is claiming that multiple projects have disappeared out of a students account. I've checked the details page of the account and do not see any projects that have been deleted. There are two in the trash folder but the Teacher says these are not the projects.

I see the two projects mentioned above in the Trash folder and none in the deleted row of the profile page. Without knowing what anything about the missing projects, I'm not sure there is anything we can do. Perhaps the user accidentally created the projects while signed out or signed into another account?

**Comment by jwzimmer** [comment]
Closing in favor of https://github.com/LLK/scratchr2/issues/4082#issuecomment-270960913.

**Comment by jwzimmer** [comment]
Reopening this because there is a difference between what was reported here and in freshdesk no. 65853 vs. what was reported in https://github.com/LLK/scratchr2/issues/4082: in 4082, the projects are not showing up in My Stuff on chromebook but do exist if they check on a Mac, whereas in the cases here, the projects are not showing up in the trashed or deleted rows at all. Both cases do so far seem to involve weird stuff with project saving and chromebook, as does https://github.com/LLK/scratch-flash/issues/1218#issue-183527295, so they could end up to be symptoms of the same issue, but for the sake of keeping straight what issue is being investigated where and what the repro should involve, I'm keeping them divided into 3 separate cases for now.

**Weird issues with chromebook & saving/ persistence of projects**

(From this ticket in https://github.com/LLK/scratch-flash/issues/1218#issue-183527295)
- Projects moving from My Stuff to Trash mysteriously
- Remixes not saving, or disappearing mysteriously

(From https://github.com/LLK/scratchr2/issues/4082)
- My Stuff being empty on Chromebook but not Mac

(From https://github.com/LLK/scratch-flash/issues/1249)
- Projects disappearing with no Trashed or Deleted projects showing up

**Comment by jwzimmer** [comment]
Here's what seems to be another example. The user mentioned in freshdesk no. 69595 doesn't have any deleted or trashed projects listed on the admin view > details page, but the teacher recounts having seen other projects. Note this is not a student account.
> [I have a student] that lost all of his projects. When I looked into his folder, there is only 1 project and it only has a baseball and nothing else.

**Comment by jwzimmer** [comment]
Another user reported this in freshdesk no. 71900.

They described a project called Morse Code Learning, which I don't see in Unshared or Trashed projects. They don't have any projects in Deleted.

**Comment by jwzimmer** [comment]
Reported in freshdesk no. 79547:
> I have a ... student who has been working on a project and it and other attempts were not shared. When she logged back in to continue working on her project everything from her account was gone. I suggested that she start over and about 10 mins after starting over her new game and any information was deleted again. We left it for a week or so and today she logged back in and there is still nothing in her account.

**Comment by jwzimmer** [comment]
From #1221:

Separated issue originally documented in https://github.com/LLK/scratchr2/issues/4133.

There is not enough information to know for sure what happened in this case, as far as I can tell, but I am hoping to get more info from the teacher, and since this would be an extremely serious bug if it did happen as they are reporting it, I thought it was worth tracking.

From freshdesk no. 59152:

> The one student who I have on record as having lost a game (the project was not in his "my stuff" folder at all) was in my  [see freshdesk ticket] class with the username [see freshdesk ticket] 
> It would have been on September 30th that the game was lost. I don't remember the name of the game in question, but he had a start screen, a short intro featuring scratch cat, the detailed bat and the banana bunch that led into the 1st level, that's all he had done up to the point the game was lost.

I looked for the user in the db and only found 5 projects associated with them, none of which had `delbyusr` status as they did in #1148 (which multiple people have encountered). 1 was in Trash, 1 was shared, and 3 were unshared. When I looked at those projects in the UI, none had any of the characteristics described by the teacher.

**TL;DR if a project vanished, with no record in the db, that is obviously very serious. I do not know for sure if the that is what happened - it's possible the student changed the project a lot, or made it in another account, or that the project didn't get saved (which would also be a problem).**

**Comment by jwzimmer** [comment]
Reported in freshdesk no. 81087:
> so all my work on scratch just deleted. i haven't been on scratch for a few weeks and i was just wondering if there is a way to get back my work or is it gone.

Metadata:
- User: (see ticket)
- Operating System: Windows NT 10.0
- Browser: Chrome 58.0.3029.96
- Flash Version: Flash 25.0 (release 0)

When I view this user in the admin panel or look them up in the db, they seem to have no projects at all.

**Comment by jwzimmer** [comment]
At this point, since several people have reported this, I think we should spend at least some time looking into it to see if we can find evidence of projects disappearing, so I removed this from the backlog.

**Comment by jwzimmer** [comment]
I looked in recent project IDs to find any missing ones. 160002998, 160002852, 160002791, 160002620 and 160002688 are (the only IDs) missing in the range 159950000 < id < 160130495. For those IDs, the project JSON 404's, so there's no evidence the projects ever got created in any way. @rschamp said if the project had unintentionally been deleted from the db, there should still be a project JSON for it, since we don't have a mechanism to perform both of those deletions at once. So for now, we have no tangible evidence of this occurring.

**Comment by jwzimmer** [comment]
A user reported their project disappearing in freshdesk no. 82304. They have no trashed or deleted projects.

**Comment by jwzimmer** [comment]
A user reported their project disappearing in freshdesk no. 90745. There are no deleted projects for the user in the db (only one project visible on the site).
> My son, userid (see ticket) has been using scratch for a few years. Today, when he signed in, he found that all his projects had vanished. When he last accessed his account during the summer, around August 10th, he still had all his projects. Was there a glitch in the system that wiped out everything in his account or did someone delete everything he had worked on deliberately?

Metadata:
- User: (see ticket)
- Operating System: MacOS Macintosh X 10.10
- Browser: Firefox 50.0
- Flash Version: Flash 26.0 (release 0)
